koku/masu/external/downloader/azure/azure_service.py

```diff
@@ -170,24 +187,39 @@ def download_file(
         destination: str = None,
         suffix: str = AzureBlobExtension.csv.value,
         ingress_reports: list[str] = None,
+        offset: int = None,
+        length: int = None,
     ) -> str:
-        """Download the file from a given storage container."""
+        """
+        Download the file from a given storage container. Supports both CSV and GZIP formats.
+        """
 
         if not ingress_reports:
             cost_export = self.get_file_for_key(key, container_name)
             key = cost_export.name
 
+            if key.endswith(AzureBlobExtension.gzip.value):
+                suffix = AzureBlobExtension.gzip.value
+            else:
+                suffix = AzureBlobExtension.csv.value
+
         file_path = destination
         if not destination:
             temp_file = NamedTemporaryFile(delete=False, suffix=suffix)
             temp_file.close()
             file_path = temp_file.name
 
         try:
-            blob_client = self._cloud_storage_account.get_blob_client(container_name, key)
+            blob_client = self._cloud_storage_account.get_blob_client(container=container_name, blob=key)
             with open(file_path, "wb") as blob_download:
-                blob_download.write(blob_client.download_blob().readall())
-        except (AdalError, AzureException, ClientException, OSError) as error:
+                if offset is not None and length is not None:
```

Comment:
I'm not sure I understand this part or why its needed? The initial download should not be a problem or need breaking up.